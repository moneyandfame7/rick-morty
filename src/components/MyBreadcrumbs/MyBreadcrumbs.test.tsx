import { render, screen } from "@testing-library/react";
import { MyBreadcrumbs } from "./index";
import { MemoryRouter } from "react-router";

// TODO: спросить как тут делать
describe("MyBreadcrumbs should render correctly", () => {
  let component: any;

  it("home route", () => {
    givenComponent();

    const label = screen.getByText("Home");

    expect(label).toBeInTheDocument();
    thenItRendersProperly();
  });

  it("unknown route", () => {
    givenComponent("/test/route/228");

    const label = screen.getByText("Information page");

    expect(label).toBeInTheDocument();
    thenItRendersProperly();
  });

  it("character route", () => {
    givenComponent("/character?page=1");

    const label = screen.getByText("Characters");

    expect(label).toBeInTheDocument();
    thenItRendersProperly();
  });

  it("should render correctly episode route", () => {
    givenComponent("/episode?page=1");

    const label = screen.getByText("Episodes");

    expect(label).toBeInTheDocument();
    thenItRendersProperly();
  });

  function givenComponent(route: string = "/") {
    component = render(
      <MemoryRouter initialEntries={[route]}>
        <MyBreadcrumbs />
      </MemoryRouter>
    );
  }

  function thenItRendersProperly() {
    expect(component).toMatchSnapshot();
  }
});
