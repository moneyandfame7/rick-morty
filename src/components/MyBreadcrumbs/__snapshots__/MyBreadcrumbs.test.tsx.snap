// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyBreadcrumbs should render correctly should correctly display labels on character page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        aria-label="breadcrumb"
        class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-1wdhu18-MuiTypography-root-MuiBreadcrumbs-root"
      >
        <ol
          class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
        >
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              class=""
              href="/"
            >
              <span
                class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-1oohpub-MuiChip-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault css-ptiqhd-MuiSvgIcon-root"
                  data-testid="HomeIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                  />
                </svg>
                <span
                  class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                >
                  Home
                </span>
              </span>
            </a>
          </li>
          <li
            aria-hidden="true"
            class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
              data-testid="NavigateNextIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </li>
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              aria-current="page"
              class="active"
              href="/character?page=1"
            >
              <span
                class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-15afu26-MuiChip-root"
              >
                <span
                  class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                >
                  Characters
                </span>
              </span>
            </a>
          </li>
        </ol>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      aria-label="breadcrumb"
      class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-1wdhu18-MuiTypography-root-MuiBreadcrumbs-root"
    >
      <ol
        class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
      >
        <li
          class="MuiBreadcrumbs-li"
        >
          <a
            class=""
            href="/"
          >
            <span
              class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-1oohpub-MuiChip-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault css-ptiqhd-MuiSvgIcon-root"
                data-testid="HomeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                />
              </svg>
              <span
                class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
              >
                Home
              </span>
            </span>
          </a>
        </li>
        <li
          aria-hidden="true"
          class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
            data-testid="NavigateNextIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
        </li>
        <li
          class="MuiBreadcrumbs-li"
        >
          <a
            aria-current="page"
            class="active"
            href="/character?page=1"
          >
            <span
              class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-15afu26-MuiChip-root"
            >
              <span
                class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
              >
                Characters
              </span>
            </span>
          </a>
        </li>
      </ol>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`MyBreadcrumbs should render correctly should only display the "home" label on the main page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        aria-label="breadcrumb"
        class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-1wdhu18-MuiTypography-root-MuiBreadcrumbs-root"
      >
        <ol
          class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
        >
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              <span
                class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-1oohpub-MuiChip-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault css-ptiqhd-MuiSvgIcon-root"
                  data-testid="HomeIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                  />
                </svg>
                <span
                  class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                >
                  Home
                </span>
              </span>
            </a>
          </li>
        </ol>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      aria-label="breadcrumb"
      class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-1wdhu18-MuiTypography-root-MuiBreadcrumbs-root"
    >
      <ol
        class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
      >
        <li
          class="MuiBreadcrumbs-li"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            <span
              class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-1oohpub-MuiChip-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault css-ptiqhd-MuiSvgIcon-root"
                data-testid="HomeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                />
              </svg>
              <span
                class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
              >
                Home
              </span>
            </span>
          </a>
        </li>
      </ol>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`MyBreadcrumbs should render correctly should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        aria-label="breadcrumb"
        class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-1wdhu18-MuiTypography-root-MuiBreadcrumbs-root"
      >
        <ol
          class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
        >
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              <span
                class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-1oohpub-MuiChip-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault css-ptiqhd-MuiSvgIcon-root"
                  data-testid="HomeIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                  />
                </svg>
                <span
                  class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                >
                  Home
                </span>
              </span>
            </a>
          </li>
        </ol>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      aria-label="breadcrumb"
      class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-1wdhu18-MuiTypography-root-MuiBreadcrumbs-root"
    >
      <ol
        class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
      >
        <li
          class="MuiBreadcrumbs-li"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            <span
              class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-1oohpub-MuiChip-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault css-ptiqhd-MuiSvgIcon-root"
                data-testid="HomeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                />
              </svg>
              <span
                class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
              >
                Home
              </span>
            </span>
          </a>
        </li>
      </ol>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`MyBreadcrumbs should render correctly when the route is unknown, should not display others label 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        aria-label="breadcrumb"
        class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-1wdhu18-MuiTypography-root-MuiBreadcrumbs-root"
      >
        <ol
          class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
        >
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              class=""
              href="/"
            >
              <span
                class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-1oohpub-MuiChip-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault css-ptiqhd-MuiSvgIcon-root"
                  data-testid="HomeIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                  />
                </svg>
                <span
                  class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                >
                  Home
                </span>
              </span>
            </a>
          </li>
          <li
            aria-hidden="true"
            class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
              data-testid="NavigateNextIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </li>
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              aria-current="page"
              class="active"
              href="/unknown/123123/5432167/12379/sdfgsdfg"
            >
              <span
                class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-1oohpub-MuiChip-root"
              >
                <span
                  class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                >
                  Information page
                </span>
              </span>
            </a>
          </li>
        </ol>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      aria-label="breadcrumb"
      class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-1wdhu18-MuiTypography-root-MuiBreadcrumbs-root"
    >
      <ol
        class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
      >
        <li
          class="MuiBreadcrumbs-li"
        >
          <a
            class=""
            href="/"
          >
            <span
              class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-1oohpub-MuiChip-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault css-ptiqhd-MuiSvgIcon-root"
                data-testid="HomeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                />
              </svg>
              <span
                class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
              >
                Home
              </span>
            </span>
          </a>
        </li>
        <li
          aria-hidden="true"
          class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
            data-testid="NavigateNextIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
        </li>
        <li
          class="MuiBreadcrumbs-li"
        >
          <a
            aria-current="page"
            class="active"
            href="/unknown/123123/5432167/12379/sdfgsdfg"
          >
            <span
              class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-1oohpub-MuiChip-root"
            >
              <span
                class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
              >
                Information page
              </span>
            </span>
          </a>
        </li>
      </ol>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
